From 5183590eadeaa5836e55e924ded8bf25220c03a8 Mon Sep 17 00:00:00 2001
From: Lou Viannay <lou.viannay@octavesolution.com>
Date: Tue, 31 Mar 2026 07:48:27 +0800
Subject: [PATCH 2/4] defconfig: enable WATCHDOG_HANDLE_BOOT_ENABLED

Kick any bootloader-armed watchdog during kernel boot to prevent warm
reset failures on boards where the bootloader arms the watchdog but
does not expect the kernel to re-arm it.

Signed-off-by: Blue Cuenca <blue.cuenca@gmail.com>
---
 arch/arm64/configs/longanpi_3h_defconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/configs/longanpi_3h_defconfig b/arch/arm64/configs/longanpi_3h_defconfig
index 11534851f..bab674109 100644
--- a/arch/arm64/configs/longanpi_3h_defconfig
+++ b/arch/arm64/configs/longanpi_3h_defconfig
@@ -821,7 +821,7 @@ CONFIG_SENSORS_PWM_FAN=m
 CONFIG_THERMAL=y
 CONFIG_SUN8I_THERMAL=y
 CONFIG_WATCHDOG=y
-# CONFIG_WATCHDOG_HANDLE_BOOT_ENABLED is not set
+CONFIG_WATCHDOG_HANDLE_BOOT_ENABLED=y
 CONFIG_SUNXI_WATCHDOG=y
 CONFIG_MFD_AXP20X_I2C=y
 CONFIG_MFD_SUN6I_PRCM=y
-- 
2.17.1

